File: ontrack-git/src/main/java/net/nemerosa/ontrack/git/model/GitCommit.java
package net.nemerosa.ontrack.git.model;

import lombok.Data;

import java.time.LocalDateTime;

@Data
public class GitCommit implements Comparable<GitCommit> {

    private final String id;
    private final String shortId;
    private final GitPerson author;
    private final GitPerson committer;
    private final LocalDateTime commitTime;
    private final String fullMessage;
    private final String shortMessage;

    @Override
    public int compareTo(GitCommit o) {
        return this.commitTime.compareTo(o.commitTime);
    }
}

File: ontrack-git/src/main/java/net/nemerosa/ontrack/git/model/GitCommit.kt
package net.nemerosa.ontrack.git.model

import java.time.LocalDateTime

class GitCommit(
        val id: String,
        val shortId: String,
        val author: GitPerson,
        val committer: GitPerson,
        val commitTime: LocalDateTime,
        val fullMessage: String,
        val shortMessage: String
) : Comparable<GitCommit> {
    override fun compareTo(other: GitCommit): Int {
        return this.commitTime.compareTo(other.commitTime)
    }
}
